### PR TITLE
Fix storage_account data source documentation

### LINKED ISSUE #1316: Storage account kind documentation
The documentation don't mention StorageV2 kind. I needed to go through the issues/prs and the code to know how to use StorageV2 as a kind for my storage account.

In the [docs](https://www.terraform.io/docs/providers/azurerm/d/storage_account.html#account_kind) we have 

>account_kind - Defines the Kind of account, either BlobStorage or Storage.

Where is missing the StorageV2 option.

### PR DESCRIPTION
**Pull request by katbyte**
Fixes #1316

### CODE REVIEW

File: website/docs/d/storage_account.html.markdown [hunk 1]
```diff
@@ -35,13 +35,13 @@ output "storage_account_tier" {
 
 * `location` - The Azure location where the Storage Account exists
 
-* `account_kind` - Defines the Kind of account, either `BlobStorage` or `Storage`.
```
Comment (metacpp) [inline website/docs/d/storage_account.html.markdown]: #1316 raised the concern on missed option of StorageV2. Is it correct that we directly delete all the options in the documentation? Then where will user find these options? Shall we provide a link ?

File: website/docs/d/storage_account.html.markdown [hunk 2]
```diff
@@ -35,13 +35,13 @@ output "storage_account_tier" {
...
+* `account_kind` - The Kind of account.
 
-* `account_tier` - Defines the Tier of this storage account.
+* `account_tier` - The Tier of this storage account.
 
-* `account_replication_type` - Defines the type of replication used for this storage account.
+* `account_replication_type` - The type of replication used for this storage account.
 
-* `access_tier` - Defines the access tier for `BlobStorage` accounts.
```
Comment (metacpp) [inline website/docs/d/storage_account.html.markdown]: We should remove "Defines" in all fields' description, is it a documentation style? Do we need to update the other documents?

File: website/docs/d/storage_account.html.markdown (hunk 1 above)
Comment (katbyte) [reply website/docs/d/storage_account.html.markdown]: This is a datasource, not a resource so we don't to specify valid values. 

There is an argument to be said for listing possible values, however currently most of the documentation for datasources don't so for consistencies sake I have removed them.

File: website/docs/d/storage_account.html.markdown (hunk 2 above)
Comment (katbyte) [reply website/docs/d/storage_account.html.markdown]: As mentioned above this is a datasource, so it doesn't define the access tier, just returns what the access tier is.

**Pull request merged**